packages/hint-no-broken-links/src/hint.ts

```diff
@@ -203,7 +203,17 @@ export default class NoBrokenLinksHint implements IHint {
                     return requester
                         .get(fullURL)
                         .then((value: NetworkData) => {
-                            return handleSuccess(value, fullURL, element);
+                            if (element.nodeName !== 'LINK') {
```

Comment:
I think I'll move this logic to `handleSuccess` because it's there when we decide if something should be reported or not. What do you think of something similar to:

```ts
const isResourceHint = (element: IAsyncHTMLElement): boolean => {
   if (element.nodeName !== 'LINK') {
     return false;
  }

  // More magic here
}

const handleSuccess = (networkData: NetworkData, url: string, element: IAsyncHTMLElement) => {
   if(isResourceHint(element)) {
    return Promise.resolve();
  }

  const statusIndex = brokenStatusCodes.indexOf(
    networkData.response.statusCode
  );

  if (statusIndex > -1) {
    return context.report(url, element, `Broken link found (${brokenStatusCodes[statusIndex]} response).`);
  }

  fetchedURLs.push({ status: networkData.response.statusCode, url });
    return Promise.resolve();
  };
```

Reply:
Yes, I think it is a good idea. Having all the code in the promise chain looks very messy.